overlord/hookstate/context.go

```diff
@@ -127,3 +128,50 @@ func (c *Context) Get(key string, value interface{}) error {
 
 	return nil
 }
+
+// State returns the state contained within the context
+func (c *Context) State() *state.State {
+	return c.task.State()
+}
+
+// Cached returns the cached value associated with the provided key.
+// It returns nil if there is no entry for key.
+func (c *Context) Cached(key interface{}) interface{} {
+	contextCache := c.task.State().Cached("hook-context")
```

Comment:
we tend to use empty structs as keys for cached,

type hookContextCache{}

 they risk less clashes than strings